Replicator/Pusher+Revs.cc

```diff
@@ -240,7 +240,12 @@ namespace litecore::repl {
                 }
                 doneWithRev(rev, completed, synced);
                 switch (retry) {
-                    case kRetryNow:   retryRevs({rev}, true); break;
+                    case kRetryNow:
+                        rev->error.code = 0;
+                        rev->isWarning = 0;
+                        rev->errorIsTransient = false;
+                        retryRevs({rev}, true);
+                        break;
```

Comment:
This issue is not described in the CBSE ticket (Only pull case). Can you provide a bit info about the behavior before the change?

Reply:
Without this change, because we use the same pointer to the previous rev as we retry, if don't clear the error, after successful retry, the error is still stuck, and we won't get the completed doc count correct. There is a minor issue here -- I meant to point out after the build succeeds and invite reviews -- that by clearing the error, before it goes to the docEnded callback, the error of the first try will be cleared as well. 
Ideally, we probably should make a clone of the rev for the retry, or make a clone to send to docEnded, but this will defeat our super-efficient way passing DocumentEnded to the callback.
I feel lose an error to the callback is not big thing, because it's pending retry. If retry fails again, it will be failed doc to the docEnded callback.

Reply:
> I feel lose an error to the callback is not big thing, because it's pending retry. If retry fails again, it will be failed doc to the docEnded callback.

Not sure I understand the behavior correctly. So the first docEnd of the failure before retry with be shown as success or the docEnd will not be sent before the retry result?

Reply:
It sounds to me like: if first attempt is fail, it will be retried. Because we now clear error on retry, the error from first fail will not be shown. So if the second attempt succeeds, we will never know the error for the first attempt. If the second attempt fails, we will get that error.

Reply:
Is there a CBL ticket for this change?  If not. please create one before committing this

Reply:
That's part of the thing I need to adjust to fix the problem in the ticket. Otherwise, the bug will show in the other direction, namely the Push.

Reply:
As to @pasin 's question, @callumbirks 's understanding is correct. Basically this what happens
1. First attempt of "rev" with delta  received response from SG that shows error.
2. We push the ReplicatedRev object to docEnded queue.
3. Meanwhile, afterwards, we decided to retry without delta, using the same ReplicatedRev object, which has the error fields assigned.
4. As is done in the PR, I cleared the errors so that second attempt starts afresh.
5. Asynchronously, when we get to report status, we pop the entries from docEnded. At this time, the ReplicatedRev object has clear error fields due to (4).
If the info about the error of the first attempt is important, I probably can do something. For example, not clear the error until the response of the second attempt arrives.

Reply:
From the API point of view, the document ended shouldn't report as success if it doesn't. Even better, if it doesn't report until retry (This is probably out of scope of this PR).